components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -411,8 +411,25 @@ private void storeAccessToken(String accessToken, String consumerKey, AccessToke
                     throw new IdentityOAuth2Exception(errorMsg, e);
                 }
 
-                recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
-                        connection, userStoreDomain, retryAttempt + 1);
+                boolean isPostgreSQL = false;
+                Connection pgConnection = IdentityDatabaseUtil.getDBConnection();
+                try {
+                    if (StringUtils.containsIgnoreCase(pgConnection.getMetaData().getDriverName(), "PostgreSQL")) {
```

Comment:
Lets move this "if logic" to a separate private and return false inside the catch block of SQLException. The method name would be like isProsgreSQLDB()